.github/workflows/aws_tfhe_fast_tests.yml

```diff
@@ -133,11 +125,18 @@ jobs:
         run: |
           echo "any_changed=true" >> "$GITHUB_OUTPUT"
 
+  check-user-permission:
+    needs: should-run
+    if: ${{ always() && needs.should-run.result != 'failure' }}
```

Comment:
is the always needed ? feels like it should run as long as should run succeeds without the if ?